pkg/controller/controller_binding.go

```diff
@@ -87,6 +89,41 @@ func isServiceInstanceCredentialFailed(binding *v1alpha1.ServiceInstanceCredenti
 	return false
 }
 
+func (c *controller) doOrphanMitigation(binding *v1alpha1.ServiceInstanceCredential, toUpdate *v1alpha1.ServiceInstanceCredential, instance *v1alpha1.ServiceInstance) error {
+	if err := c.reconcileServiceInstanceCredentialDelete(binding, toUpdate, instance); err != nil {
+		return err
+	}
+	if err := c.updateServiceInstanceCredentialStatus(toUpdate); err != nil {
+		return err
+	}
```

Comment:
I'm beginning to wonder if these methods need to be idempotent (if they aren't already), since a failure would cause another mitigation attempt to occur, but potentially with some resources already deleted.